### PR TITLE
collations: cleanup all data generation

### PR DESCRIPTION
**Pull request by vmg**
## Description

I'm wrapping up this short week with a large refactoring of how we're building the collation metadata for our `collations` package. As I ported over more and more collations from MySQL over the past month, I ended piling up a bunch of different tooling under `tools/` which was ad-hoc and not particularly beautiful to do that.

I've refactored all that tooling into a single `makecolldata` tool that should be much cleaner and much easier to read and follow. It generates all the native metadata for all the collations we know about by simply running `go generate .` in the root of the `collations` package.

That's all! The actual interfaces or behavior of the `collations` package hasn't changed at all. It just now looks cleaner on the inside.

## Related Issue(s)
- https://github.com/vitessio/vitess/issues/860

## Checklist
- [ ] Should this PR be backported?
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

File: go/mysql/collations/internal/uca/layout.go [hunk 1]
```diff
@@ -0,0 +1,233 @@
+package uca
```
Comment (systay) [inline go/mysql/collations/internal/uca/layout.go:1]: license header missing

File: go/mysql/collations/collation.go [hunk 2]
```diff
@@ -22,8 +22,8 @@ import (
 	"vitess.io/vitess/go/mysql/collations/internal/charset"
 )
 
-// Generate mysqldata.go from the JSON information dumped from MySQL
-//go:generate go run ./tools/makemysqldata/
+// Generate all the metadata used for collations from the JSON data dumped from MySQL
+//go:generate go run ./tools/makecolldata/
```
[the remark this answers is not in the document]
Comment (vmg) [reply go/mysql/collations/collation.go:26]: Yeah! This does the trick.

File: go/mysql/collations/tools/makecolldata/codegen/format/rewrite.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
+// Copyright 2009 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package format
+
```
[the remark this answers is not in the document]
Comment (vmg) [reply go/mysql/collations/tools/makecolldata/codegen/format/rewrite.go:6]: This is an internal package in `gofmt`, so we cannot copy it, sadly. I had to import it like this so we can actually run formatting for our generated code with the `-simplify` flag, which is not exported by default. If we don't run with `-simplify`, all our linters will error with the generated code.

[the remark this answers is not in the document]
Comment (vmg) [reply go/mysql/collations/tools/makecolldata/codegen/format/rewrite.go]: Took care of this by actually removing the checked in `format` code and shelling out to `gofmt` directly.

**Pull request merged**